doc/packages/linux.section.md

```diff
@@ -105,3 +105,58 @@ $ make -C $dev/lib/modules/*/build M=$(pwd)/drivers/net/ethernet/mellanox module
 ```
 
 :::
+
+## Packaging out of tree kernel modules {#sec-packaging-out-of-tree-kernel-modules}
+
+It is recommended to use the build helper `buildKernelModule`. It has improved compatibility with existing update scripts like
+`nix-update` and sets some reasonable defaults. Here's an example below:
+
+```nix
+{ lib, buildKernelModule, fetchFromGitHub, kernel, kernelModuleMakeFlags, bc, nix-update-script }:
+
+buildKernelModule {
+  pname = "rtl8821cu";
+  version = "0-unstable-2025-09-10";
+
+  src = fetchFromGitHub {
+    owner = "morrownr";
+    repo = "8821cu-20210916";
+    rev = "07fa9cf0fa8b0c08920c359c725dfc250e91422b";
+    hash = "sha256-JAkh0Vnt+Hg16F2xCsFPs5SAmaS2oqdIf45L0hXN0iY=";
+  };
+
+  nativeBuildInputs = [ bc ];
+  makeFlags = kernelModuleMakeFlags;
+
+  prePatch = ''
+    substituteInPlace ./Makefile \
+      --replace-fail /lib/modules/ "${kernel.dev}/lib/modules/" \
+      --replace-fail /sbin/depmod \# \
+      --replace-fail '$(MODDESTDIR)' "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+  '';
+
+  preInstall = ''
+    mkdir -p "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+  '';
+
+  passthru.updateScript = nix-update-script { extraArgs = [ "--version=branch" ]; };
+
+  meta = {
+    description = "Realtek rtl8821cu driver";
+    homepage = "https://github.com/morrownr/8821cu-20210916";
+    license = lib.licenses.gpl2Only;
+    maintainers = [ lib.maintainers.contrun ];
+  };
+}
+```
+
+It then needs to be imported in `pkgs/top-level/linux-kernels.nix`.
+
+```nix
+  rtl8821cu = callPackage ../os-specific/linux/rtl8821cu { };
+```
+
+::: {.note}
+Kernel modules should NOT be in `by-name` scope or be imported in `pkgs/top-level/all-packages.nix`.
+Top level attributes like `kernel`, `kernelModuleMakeFlags`, `buildKernelModule` are not available there.
+:::
```

Comment:
Rather than going by what it's not, it might make more sense to give some context? I.e. that we have per-version attribute-sets such that we can build modules against misc kernel versions.